src/Transport/Topology/Topologies/EndpointOrientedTopologySectionManager.cs

```diff
@@ -87,71 +87,87 @@ public TopologySectionInternal DetermineResourcesToCreate(QueueBindings queueBin
-                if (inputQueueAddress.HasConnectionString)
+                var inputQueueAddress = addressingLogic.Apply(d, EntityType.Queue);
+
+                RuntimeNamespaceInfo[] namespaces = null;
+                if (inputQueueAddress.HasSuffix && inputQueueAddress.Suffix != defaultNameSpaceAlias) // sending to specific namespace
                 {
-                    namespaces = new[]
+                    if (inputQueueAddress.HasConnectionString)
                     {
-                        new RuntimeNamespaceInfo(inputQueueAddress.Suffix, inputQueueAddress.Suffix, NamespacePurpose.Routing, NamespaceMode.Active)
-                    };
+                        namespaces = new[]
+                        {
+                            new RuntimeNamespaceInfo(inputQueueAddress.Suffix, inputQueueAddress.Suffix, NamespacePurpose.Routing)
+                        };
+                    }
+                    else
+                    {
+                        var configured = namespaceConfigurations.FirstOrDefault(n => n.Alias == inputQueueAddress.Suffix);
+                        if (configured != null)
+                        {
+                            namespaces = new[]
+                            {
+                                new RuntimeNamespaceInfo(configured.Alias, configured.Connection, configured.Purpose)
+                            };
+                        }
+                    }
                 }
                 else
                 {
-                    var configured = namespaceConfigurations.FirstOrDefault(n => n.Alias == inputQueueAddress.Suffix);
+                    var configured = namespaceConfigurations.FirstOrDefault(n => n.RegisteredEndpoints.Contains(d, StringComparer.OrdinalIgnoreCase));
                     if (configured != null)
                     {
                         namespaces = new[]
                         {
-                            new RuntimeNamespaceInfo(configured.Alias, configured.Connection, configured.Purpose, NamespaceMode.Active)
+                            new RuntimeNamespaceInfo(configured.Alias, configured.Connection, configured.Purpose),
                         };
                     }
+                    else // sending to the partition
+                    {
+                        namespaces = namespacePartitioningStrategy.GetNamespaces(PartitioningIntent.Sending).ToArray();
+                    }
                 }
-            }
-            else // sending to the partition
-            {
-                namespaces = namespacePartitioningStrategy.GetNamespaces(PartitioningIntent.Sending).ToArray();
-            }
 
-            if (namespaces == null)
-            {
-                throw new Exception($"Could not determine namespace for destination {destination}");
-            }
+                if (namespaces == null)
+                {
+                    throw new Exception($"Could not determine namespace for destination {d}");
```

Comment:
This was done incorrectly in the past, but should we consider escaping the value with single quotes? Doesn't have to be this PR at all, just a question 🙂

Reply:
Will do